src/shapes/itext.class.js

```diff
@@ -271,7 +271,19 @@
       this._clearTextArea(ctx);
       skipRestore || ctx.restore();
     },
-
+    transform: function(ctx) {
+      var m;
+      // the contextTop layer will not be transform in group render function. so we need
+      // manual apply the group transform for render cursor or selection which used the contextTop
+      if ((this.group && !this.group._transformDone) ||
+          (this.group && this.canvas && ctx === this.canvas.contextTop)) {
+        m = this.calcTransformMatrix();
+      }
+      else {
+        m = this.calcOwnMatrix();
+      }
+      ctx.transform(m[0], m[1], m[2], m[3], m[4], m[5]);
+    },
```

Comment:
i wonder if we can generalize this in the main Object class.
Does it make sense to duplicate it?
This is the standard function
```
    /**
     * Transforms context when rendering an object
     * @param {CanvasRenderingContext2D} ctx Context
     */
    transform: function(ctx) {
      var m;
      if (this.group && !this.group._transformDone) {
        m = this.calcTransformMatrix();
      }
      else {
        m = this.calcOwnMatrix();
      }
      ctx.transform(m[0], m[1], m[2], m[3], m[4], m[5]);
    },
```

it can be changed in the Object class in this way:

```js
    /**
     * Transforms context when rendering an object
     * @param {CanvasRenderingContext2D} ctx Context
     */
    transform: function(ctx) {
      var needFullTransform = (this.group && !this.group._transformDone) ||
         (this.group && this.canvas && ctx === this.canvas.contextTop);
      var m = this.calcTransformMatrix(!needFullTransform);
      ctx.transform(m[0], m[1], m[2], m[3], m[4], m[5]);
    },
```

We would need at least a visual test for it of some sort anyway.